### PR TITLE
12 school assignment

### PR DESCRIPTION
**Pull request by AnuKPatel**
Hello,

Migration of assignment module is done.
Please review the code and merge.

### CODE REVIEW

File: assignment/views/homework_view.xml [hunk 1]
```diff
@@ -227,7 +226,8 @@
                 <data>
                     <xpath expr="//div[@name='buttons']" position="inside">
                         <button name="%(action_school_student_assignment_form_btn)d" type="action" class="oe_stat_button" attrs="{'invisible':[('state', 'in', ['draft','terminate','alumni','cancel'])]}" groups="school.group_school_administration,school.group_school_teacher,school.group_school_student">
-                             <img src="/assignment/static/description/Assign.png" style="height:25px;width:50px;float:left;"/>
+                             <img src="/assignment/static/description/Assign.png" alt="Logo"
+                                 style="height:25px;width:50px;float:left;"/>
                                 <lable style = "">Assignment</lable>
```
Comment (dhirendrasinh-Serpent) [inline assignment/views/homework_view.xml:231]: No need to style attribute.

File: assignment/views/homework_view.xml [hunk 2]
```diff
@@ -195,7 +194,7 @@
                     <field name="due_date"/>
                     <newline/>
                     <group expand="0" string="Group By...">
-                        <filter string="State" context="{'group_by': 'state'}" icon="terp-stock_effects-object-colorize"/>
+                        <filter string="State" name="state" context="{'group_by': 'state'}" icon="terp-stock_effects-object-colorize"/>
```
Comment (dhirendrasinh-Serpent) [inline assignment/views/homework_view.xml]: No need icon attribute in filter V12

File: assignment/views/homework_view.xml [hunk 3]
```diff
@@ -195,7 +194,7 @@
...
                         <filter string="Student" name="student_id" icon="terp-folder-orange" domain="" context="{'group_by':'student_id'}"/>
```
Comment (dhirendrasinh-Serpent) [inline assignment/views/homework_view.xml]: No need icon attribute in filter V12

File: assignment/views/homework_view.xml [hunk 4]
```diff
@@ -195,7 +194,7 @@
...
                         <filter string="Teacher" name="teacher_id" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'teacher_id'}"/>
```
Comment (dhirendrasinh-Serpent) [inline assignment/views/homework_view.xml]: No need icon attribute in filter V12

File: assignment/views/homework_view.xml [hunk 5]
```diff
@@ -195,7 +194,7 @@
...
                         <filter string="Subject" name="subject_id" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'subject_id'}"/>
```
Comment (dhirendrasinh-Serpent) [inline assignment/views/homework_view.xml]: No need icon attribute in filter V12

File: assignment/views/homework_view.xml [hunk 6]
```diff
@@ -35,11 +35,11 @@
...
-                        <filter string="Subject" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'subject_id'}"/>
-                        <filter string="Standard" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'standard_id'}"/>
-                        <filter string="Assign Date" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'assign_date'}"/>
+                        <filter name="state" string="State" context="{'group_by': 'state'}" icon="terp-stock_effects-object-colorize"/>
+                        <filter name="teacher" string="Teacher" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'teacher_id'}"/>
+                        <filter name="subject" string="Subject" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'subject_id'}"/>
+                        <filter name="standard" string="Standard" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'standard_id'}"/>
+                        <filter name="assign_date" string="Assign Date" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'assign_date'}"/>
```
Comment (dhirendrasinh-Serpent) [inline assignment/views/homework_view.xml]: No need icon attribute in filter V12

File: assignment/views/homework_view.xml [hunk 7]
```diff
@@ -35,11 +35,11 @@
...
-                        <filter string="Teacher" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'teacher_id'}"/>
-                        <filter string="Subject" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'subject_id'}"/>
-                        <filter string="Standard" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'standard_id'}"/>
-                        <filter string="Assign Date" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'assign_date'}"/>
+                        <filter name="state" string="State" context="{'group_by': 'state'}" icon="terp-stock_effects-object-colorize"/>
+                        <filter name="teacher" string="Teacher" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'teacher_id'}"/>
+                        <filter name="subject" string="Subject" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'subject_id'}"/>
+                        <filter name="standard" string="Standard" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'standard_id'}"/>
```
Comment (dhirendrasinh-Serpent) [inline assignment/views/homework_view.xml]: No need icon attribute in filter V12

File: assignment/views/homework_view.xml [hunk 8]
```diff
@@ -35,11 +35,11 @@
...
                     <newline/>
                     <group expand="0" string="Group By...">
-                        <filter string="State" context="{'group_by': 'state'}" icon="terp-stock_effects-object-colorize"/>
-                        <filter string="Teacher" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'teacher_id'}"/>
-                        <filter string="Subject" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'subject_id'}"/>
-                        <filter string="Standard" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'standard_id'}"/>
-                        <filter string="Assign Date" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'assign_date'}"/>
+                        <filter name="state" string="State" context="{'group_by': 'state'}" icon="terp-stock_effects-object-colorize"/>
+                        <filter name="teacher" string="Teacher" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'teacher_id'}"/>
+                        <filter name="subject" string="Subject" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'subject_id'}"/>
```
Comment (dhirendrasinh-Serpent) [inline assignment/views/homework_view.xml]: No need icon attribute in filter V12

File: assignment/views/homework_view.xml [hunk 9]
```diff
@@ -35,11 +35,11 @@
                     <field name="due_date"/>
                     <newline/>
                     <group expand="0" string="Group By...">
-                        <filter string="State" context="{'group_by': 'state'}" icon="terp-stock_effects-object-colorize"/>
-                        <filter string="Teacher" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'teacher_id'}"/>
-                        <filter string="Subject" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'subject_id'}"/>
-                        <filter string="Standard" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'standard_id'}"/>
-                        <filter string="Assign Date" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'assign_date'}"/>
+                        <filter name="state" string="State" context="{'group_by': 'state'}" icon="terp-stock_effects-object-colorize"/>
+                        <filter name="teacher" string="Teacher" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'teacher_id'}"/>
```
Comment (dhirendrasinh-Serpent) [inline assignment/views/homework_view.xml]: No need icon attribute in filter V12

File: assignment/views/homework_view.xml [hunk 10]
```diff
@@ -35,11 +35,11 @@
                     <field name="due_date"/>
                     <newline/>
                     <group expand="0" string="Group By...">
-                        <filter string="State" context="{'group_by': 'state'}" icon="terp-stock_effects-object-colorize"/>
-                        <filter string="Teacher" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'teacher_id'}"/>
-                        <filter string="Subject" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'subject_id'}"/>
-                        <filter string="Standard" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'standard_id'}"/>
-                        <filter string="Assign Date" icon="terp-stock_symbol-selection" domain="" context="{'group_by':'assign_date'}"/>
+                        <filter name="state" string="State" context="{'group_by': 'state'}" icon="terp-stock_effects-object-colorize"/>
```
Comment (dhirendrasinh-Serpent) [inline assignment/views/homework_view.xml]: No need icon attribute in filter V12

**Review by dhirendrasinh-Serpent — COMMENTED** [review]
Improve code

File: assignment/views/homework_view.xml (hunk 1 above)
Comment (AnuKPatel) [reply assignment/views/homework_view.xml:231]: Here Style is to manage the image added in smart button and it's needed.

**Comment by AnuKPatel** [comment]
Removed icon from all the filters.

**Pull request merged**